src/borg/archive.py

```diff
@@ -1360,16 +1360,18 @@ def valid_item(obj):
             sort_by = sort_by.split(',')
             if any((first, last, prefix)):
                 archive_infos = self.manifest.archives.list(sort_by=sort_by, prefix=prefix, first=first, last=last)
+                if not archive_infos:
+                    logger.warning('--first/--last/--prefix did not match any archives')
```

Comment:
i have a bit more here to handle less-than-expected cases for --first and --last.